internal/snmp/field.go

```diff
@@ -94,6 +96,10 @@ func (f *Field) Init(tr Translator) error {
 // fieldConvert converts from any type according to the conv specification
 func (f *Field) Convert(ent gosnmp.SnmpPDU) (v interface{}, err error) {
 	if f.Conversion == "" {
+		// OctetStrings may contain hex data that needs its own conversion
+		if ent.Type == gosnmp.OctetString && !utf8.ValidString(string(ent.Value.([]byte)[:])) {
+			return hex.EncodeToString(ent.Value.([]byte)), nil
+		}
 		if bs, ok := ent.Value.([]byte); ok {
 			return string(bs), nil
 		}
```

Comment:
This part could actually be removed, since this conversion is eventually done in `convertField`: https://github.com/influxdata/telegraf/blob/d42e407ab7a2e12ff252beaa0c72509059298f48/metric/metric.go#L306-L307

Reply:
I'll talk to the team about it...

Reply:
We agreed that removing this in the SNMP part is fine. So please go ahead and remove the conversion here.

Reply:
Nevermind, that caused problems when the value appeared to be a metric tag.

Reply:
Ok, so we keep it...